src/MC/fix_gcmc.cpp

```diff
@@ -2281,16 +2314,24 @@ double FixGCMC::energy_full()
         delz = x[i][2] - x[j][2];
         rsq = delx*delx + dely*dely + delz*delz;
 
-        if (rsq < overlap_cutoffsq) {
+        if (overlap_flag && rsq < overlap_cutoffsq) {
           overlaptest = 1;
           break;
         }
+        if (max_sep_flag && rsq > max_sepsq) {
+          max_septest = 1;
+          break;
+        }
```

Comment:
This logic seems wrong.  A connected cluster will naturally have many occurrences of `rsq > max_sepsq`.

Reply:
Thank you for a thorough review. I very much agree with you that it is better to look at this function as a way to prevent isolated atoms, rather than enforce atoms to be in a cluster. To answer your questions, we need to consider what would be the typical use case of this function. Usually, one wants to prevent isolated atoms when only the cluster(s) are of interest, which is also my case. Then it makes sense to reject moves that split up a cluster in addition to rejecting moves where isolated atoms are added. I didn't really think about this issue before you mentioned it. One possible way of doing this would be to count the number of clusters and check whether or not the number of clusters is conserved when performing a move. However, I don't know if this would be the most efficient way.

Another major issue with this approach, is that most of the swap moves will be rejected. It is very unlikely that a particle will be inserted inside a cluster, and the zone around a cluster where a particle is allowed to be inserted is typically very narrow. To overcome this issue, modified Monte Carlo sampling techniques, like aggregation-volume-bias Monte Carlo (AVBMC) [1] and even energy-biased aggregation-volume-bias Monte Carlo (EB-AVBMC) [2] have been suggested. Do you think it would be possible to implement these kinds of techniques in LAMMPS, or is it out of the scope of the package? 

Anyway, I realize that there are several issues with this implementation that I were not aware of. I may come back with a more thought-through PR later, but for now on I think it's better to not merge this PR.

Thank you for your help!

[1] https://pubs.acs.org/doi/10.1021/jp709693g
[2] https://pubs.acs.org/doi/10.1021/acs.jctc.5b00466